galsim/sensor.py

```diff
@@ -348,22 +348,35 @@ def _calculate_diff_step(self):
         PixelSize = self.config['PixelSizeX']
         SensorThickness = self.config['SensorThickness']
         ChannelStopWidth = self.config['ChannelStopWidth']
+        FieldOxideTaper = self.config["FieldOxideTaper"]        
         Vbb = self.config['Vbb']
         Vparallel_lo = self.config['Vparallel_lo']
         Vparallel_hi = self.config['Vparallel_hi']
+        qfh = self.config["qfh"]        
         CCDTemperature = self.config['CCDTemperature']
         # This calculates the diffusion step size given the detector
         # parameters.  The diffusion step size is the mean radius of diffusion
         # assuming the electron propagates the full width of the sensor.
         # It depends on the temperature, the sensor voltages, and
         # the diffusion_factor parameter.
+        # The diffusion sigma will be scaled in Silicon.cpp
+        # depending on the conversion depth
 
         # Set up the diffusion step size at the operating temperature
-        HiPhases = CollectingPhases
-        LoPhases = NumPhases - CollectingPhases
-        Vdiff = (LoPhases * Vparallel_lo + HiPhases * Vparallel_hi) / NumPhases - Vbb
+        # First, calculate the approximate front side voltage
+        VChannelStop = qfh # near zero
+        VCollect = Vparallel_hi + 12.0 # Estimate from simulation
+        VBarrier = Vparallel_lo + 15.0 # Estimate from simulation
+        ChannelStopRegionWidth = 2.0 * (ChannelStopWidth / 2.0 + FieldOxideTaper)
+        ChannelStopRegionArea = ChannelStopRegionWidth * PixelSize
+        CollectArea = (PixelSize - ChannelStopRegionWidth) * PixelSize * CollectingPhases / NumPhases 
+        BarrierArea = (PixelSize - ChannelStopRegionWidth) * PixelSize * (NumPhases - CollectingPhases) / NumPhases 
+        Vfront = (ChannelStopRegionArea * VChannelStop + CollectArea * VCollect + BarrierArea * VBarrier) / (PixelSize**2)
+        # Then, the total voltage across the silicon 
+        Vdiff = max(Vfront - Vbb, 1.0) # This just makes sure that Vdiff is always > 1.0V
+        MobilityFactor = 0.27 # This is the factor from Green et.al.        
         # 0.026 is kT/q at room temp (298 K)
-        diff_step = np.sqrt(2 * 0.026 * CCDTemperature / 298.0 / Vdiff) * SensorThickness
+        diff_step = np.sqrt(2 * 0.026 * CCDTemperature / 298.0 / Vdiff / MobilityFactor) * SensorThickness
```

Comment:
I'm trying to connect this formula to what is in your slides.  I'm pretty sure I followed everything except the factor of 2 here.  Is there a 2 missing on your slide, or am I missing something about why that belongs here?

Reply:
The factor of 2 belongs there.  It was missing from the slides.  I posted a new version on the SAWG site.  Thanks for catching that.  The GalSim code always had the factor of 2.

Reply:
Also here are the updated slides.
[SAWG_Fe55_10Aug18.pdf](https://github.com/GalSim-developers/GalSim/files/2278967/SAWG_Fe55_10Aug18.pdf)